### ISSUE TITLE
All public static methods getting registered as functions

### ISSUE THREAD

**Issue by alainbryden** [Excel-DNA/ExcelDna#44]
Hey Govert,

I recently updated from 0.32 to 0.33 to see if I can get that sweet Intellisense working. In doing so I noticed it's now registering all public static methods in my assembly as excel functions.

My expectation (and I believe the prior behaviour) was that only methods with the `[ExcelFunction()]` attribute or similar would get registered. Is there an easy way to get back to that?

**Comment by govert** [comment]
Hi Alain,

This aspect did not change in version 0.33 - all the public static functions were registered with previous versions of Excel-DNA too.

You can control this by adding an `ExplicitExports='true'` attribute to your .dna file:

```
  <DnaLibrary Name="My Add-In" RuntimeVersion="v4.0" >
    <ExternalLibrary Path="MyLibrary.dll" Pack="true" ExplicitExports="true" />
  </DnaLibrary>
```

Then only functions and macros marked with `[ExcelFunction]` or `[ExcelCommand]` will be registered.

Progress on the IntelliSense project is still slow - there is certainly not enough to try out yet. But I should get some time for it over the holiday season.

-Govert

**Comment by alainbryden** [comment]
Great, thanks for the fast turnaround.

Perhaps I only just noticed because of Registration warnings being added?

I got an error "Registration [Error] Repeated function name: 'GetErrMessage' - previous registration will be overwritten." for the first time after updating, as this was the name of a public static method in two custom exception classes of mine. Never noticed the extra registrations before that.
